R/proportional_hazards-glmnet.R

```diff
@@ -192,11 +192,21 @@ print._coxnet <- function(x, ...) {
 
 # prediction --------------------------------------------------------------
 
-coxnet_predict_pre <- function(new_data, object) {
-  parsnip::.convert_form_to_xy_new(
-    object$preproc$coxnet,
-    new_data,
-    composition = "matrix")$x
+coxnet_prepare_x <- function(new_data, object) {
```

Comment:
general context: glmnet has a special treatment in censored to allow for stratification: glmnet puts stratification on the response. In the parsnip machinery, this works for fitting but breaks at prediction time because the preprocessing of `new_data` would only include the columns on the right-hand side of the formula. So we get data (and here new_data) through the parsnip machinery and then do the conversion to the right matrix ourselves.